### PR TITLE
Add debian-10-armhf build

### PR DESCRIPTION
**Pull request by stahnma**
I wanted Raspberry Pi agents to be modern. This change allows the
puppet-runtime to build on debian-10-armhf on a direct compile (not cross)
because modern arm systems aren't totally terrible in terms of speed.

### CODE REVIEW

**Comment by stahnma** [comment]
Please look over the commit on `runtime.sh` with extra care. I worked for a quite a while to try to come with a better solution there and it just didn't work. It looks to me like there's a combination of not using pl-build-tools, not using internal mirrors, and not cross compiling that caused a weird state, but I'm not positive on it. I solved the problem, but I might have created new ones, so this is the one thing in this entire patch set that could use some better eyes.

**Comment by stahnma** [comment]
On the Azure failure, I don't think it's related to any change I made.

File: resources/files/runtime/runtime.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ else
     LIBDIR="$1"
   else
     echo "$1 does not exist" >2
-    exit 1
+    exit 0
```
Comment (gimmyxd) [inline resources/files/runtime/runtime.sh:19]: I think that we bypassed this by not running this sh on certain platforms. I don't think this change should affect any of existing platforms

File: configs/components/openssl-1.1.1.rb [hunk 2]
```diff
@@ -78,6 +78,9 @@
       target = 'linux-ppc64le'
     elsif platform.architecture =~ /64$/
       target = 'linux-x86_64'
+    # Not a cross-compile for debian-10-arm
+    elsif platform.name =~ /debian-10-arm/
```
Comment (GabrielNagy) [inline configs/components/openssl-1.1.1.rb:82]: I know this has been done in a similar way for Debian 8, but it'd be more future-proof to change this `if` to something like `platform.architecture =~ /armhf/`.

File: configs/projects/_shared-agent-settings.rb [hunk 3]
```diff
@@ -99,7 +99,7 @@
 platform_triple = "ppc64le-redhat-linux" if platform.architecture == "ppc64le"
 platform_triple = "powerpc64le-suse-linux" if platform.architecture == "ppc64le" && platform.name =~ /^sles-/
 platform_triple = "powerpc64le-linux-gnu" if platform.architecture == "ppc64el"
-platform_triple = "arm-linux-gnueabihf" if platform.name == 'debian-8-armhf'
+platform_triple = "arm-linux-gnueabihf" if platform.name =~ /debian-[\d?\d]-armhf/
```
Comment (GabrielNagy) [inline configs/projects/_shared-agent-settings.rb:102]: Same as my other comment, we can use `platform.architecture =~ "armhf"` as a condition.

**Review by GabrielNagy — COMMENTED** [review]
Just some minor comments, but should be good to merge otherwise!

**Review by stahnma — COMMENTED** [review]
Changes made and squashed into original arm commit.

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**